src/components/carousel/CarouselList.vue

```diff
@@ -0,0 +1,143 @@
+<template>
+  <section id="carousel">
+    <div class="container">
+      <div class="carousel-wrap" ref="carouselWrap">
+        <div class="carousel-list" ref="carouselList">
+          <carousel-item
+            v-for="(item, index) in carouselList"
+            :key="item.id"
+            :item="item"
+            :index="index"
+            ref="carouselItem"
+          ></carousel-item>
+        </div>
+      </div>
+      <div class="carousel__navigation">
+        <div class="carousel__pagination">
+          <span
+            v-for="(item, index) in carouselList"
+            :key="item.id"
+            :class="{
+              'pagination-bullet pagination-bullet--active': index === 0,
+              'pagination-bullet': index !== 0,
+            }"
+            ref="paginationItem"
+            @click="moveCarouselOnPaginationHandler(index)"
+          ></span>
+        </div>
+        <div class="carousel__buttons">
+          <button
+            type="button"
+            class="btn carousel-prev"
+            @click="moveCarouselHandler('back')"
+            :disabled="setAtrDisabledToBack"
+          ></button>
+          <button
+            type="button"
+            class="btn carousel-next"
+            @click="moveCarouselHandler('forward')"
+            :disabled="setAtrDisabledToFoeward"
+          ></button>
+        </div>
+      </div>
+    </div>
+  </section>
+</template>
+<script>
+import t from "vue-types";
+import CarouselItem from "./CarouselListItem.vue";
+export default {
+  props: {
+    carouselList: t.Array,
```

Comment:
```suggestion
    carouselList: t.array.def([]),
```
Just to be sure and have a safe fallback.